### DISCUSSION TITLE
Error message: "redis: connection pool timeout"

### DISCUSSION THREAD

**Discussion by hongfangao** [redis/go-redis#1916] [Q&A]
Hello, at the time when I start my server, the connection to redis is OK, but after a period of time(the redis server is not operated), the request to redis will be error, and the error message is "redis: connection pool timeout". So can you help me find what's wrong with my project?
I used the redis object as follows:

```
// global variable
var myClient *redis.Client
 
// init myClient
init() {
    ...
}

... 

// use:
res, err := myClient.Get("key").Result()
```

I guess that I should use as follows, but not sure: 

```
// use:
res, err := myClient.Conn().Get("key").Result()
```

Can you help me?

=================
Another problem is that: why "myClient.Get("key")" and "myClient.Conn().Get("key")" are both legal? What's the difference of them?

Thank you!

**Comment by vmihailenco** [comment]
>So can you help me find what's wrong with my project?

See https://redis.uptrace.dev/guide/performance.html#redis-connection-pool-timeout

>Another problem is that: why "myClient.Get("key")" and "myClient.Conn().Get("key")" are both legal? What's the difference of them?

https://redis.uptrace.dev/guide/server.html#conn
